providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -1266,8 +1285,16 @@ static void *ec_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
     ret = ec_gen_assign_group(ec, gctx->gen_group);
 
     /* Whether you want it or not, you get a keypair, not just one half */
-    if ((gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) != 0)
-        ret = ret && EC_KEY_generate_key(ec);
+    if ((gctx->selection & OSSL_KEYMGMT_SELECT_KEYPAIR) != 0) {
```

Comment:
Potentially the man page provider-keymgmt(7) needs some updates to give some greater guidance on what impact the `selection` parameter has and what the expected behaviour is for different selections.

Reply:
Yeah, good point.